pkg/apis/config/v1beta1/types.go

```diff
@@ -88,6 +88,12 @@ type TargetLoadPackingArgs struct {
 	DefaultRequestsMultiplier *string `json:"defaultRequestsMultiplier,omitempty"`
 	// Node target CPU Utilization for bin packing
 	TargetUtilization *int64 `json:"targetUtilization,omitempty"`
+	// Types of client to be used: k8s, prometheus, load-watcher
```

Comment:
Seems like it's enum. If so, prefer to define an alias type:

```go
type MetricProviderType string

const (
    Kubernetes MetricProviderType = "kubernetes"
    Prometheus MetricProviderType = "prometheus"
    LoadWather MetricProviderType = "load-wather"
)
```

And used in the API (name as MetricProvider~~Type~~):

```go
MetricProvider MetricProviderType `json:"metricProvider,omitempty"`
```

Reply:
Also, can we add validation logic to ensure user input is correct?

Reply:
Updated accordingly.